docs/core/diagnostics/metrics-instrumentation.md

```diff
@@ -0,0 +1,463 @@
+        };
+    }
+}
+```
+
+When run with dotnet-counters as before the result is:
+
+```dotnetcli
+Press p to pause, r to resume, q to quit.
+    Status: Running
+
+[HatCo.HatStore]
+    orders-pending
+        Country=Italy                                  6
+        Country=Mexico                                 1
+        Country=Spain                                  3
+```
+
+#### Best Practices
+
+- Although the API allows any object to be used as the tag value, numeric types and strings are anticipated by collection tools. Other types may or may not be
+  supported by a given collection tool.
+
+- Beware of having very large or unbounded combinations of tag values being recorded in practice. Although the .NET API implementation can handle it, collection tools will
+  likely allocate storage for metric data associated with each tag combination and this could become very large. For example it is fine if HatCo has 10 different
+  hat colors and 25 hat sizes for up to 10*25=250 sales totals to track. However if HatCo added a 3rd tag that is a CustomerID for the sale and they sell to 100
+  million customers world-wide now there are now likely to be billions of different tag combinations being recorded. Most metric collection tools will either drop data
+  to stay within technical limits or there can be large monetary costs to cover the data storage and processing. The implementation of each collection tool will determine
+  its limits but likely < 1K combinations for one instrument is safe, above 1K will need the collection tool to apply filtering or be engineered to operate at high scale.
+  Histogram implementations tend to use far more memory than other metrics so safe limits could be 10-100x lower. If you anticipate large number of unique tag combinations
+  then logs, transactional databases or big data processing systems may be more appropriate solutions to operate at the needed scale.
+
+- For instruments that will have very large numbers of tag combinations, prefer using a smaller storage type to help reduce memory overhead. For example storing the `short` for
+  a `Counter<short>` only occupies 2 bytes per tag combination whereas a `double` for `Counter<double>` occupies 8 bytes per tag combination.
+
+- Collection tools are encouraged to optimize for code that specifies the same set of tag names in the same order for each call to record measurements on the
+  same instrument. For high performance code that needs to call <xref:System.Diagnostics.Metrics.Counter`1.Add%2A> and <xref:System.Diagnostics.Metrics.Histogram`1.Record%2A>
+  frequently prefer using the same sequence of tag names for each call.
+
+- The .NET API is optimized to be allocation-free for <xref:System.Diagnostics.Metrics.Counter`1.Add%2A> and <xref:System.Diagnostics.Metrics.Histogram`1.Record%2A> calls
+  with three or fewer tags specified individually. To avoid allocations with larger numbers of tags, use the <xref:System.Diagnostics.TagList>. In general,
```

Comment:
```suggestion
  with three or fewer tags specified individually. To avoid allocations with larger numbers of tags, use <xref:System.Diagnostics.TagList>. In general,
```